query-engine/connectors/mongodb-query-connector/src/interface/utils.rs

```diff
@@ -14,9 +14,15 @@ pub async fn commit_with_retry(session: &mut ClientSession) -> Result<()> {
     let timeout = Instant::now();
 
     while let Err(err) = session.commit_transaction().await {
-        if (err.contains_label(UNKNOWN_TRANSACTION_COMMIT_RESULT) || err.contains_label(TRANSIENT_TRANSACTION_ERROR))
-            && timeout.elapsed() < MAX_TX_TIMEOUT_COMMIT_RETRY_LIMIT
-        {
+        // For some reason, MongoDB adds `TRANSIENT_TRANSACTION_ERROR` to errors about aborted
```

Comment:
Because of this, previous implementation waited for 12 seconds (`MAX_TX_TIMEOUT_COMMIT_RETRY_LIMIT`) when attempting to commit on an aborted transaction.